src/implementations/fetch_issues_from_github.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+require 'bas/bot/base'
+require 'bas/shared_storage/postgres'
+require 'bas/utils/github/octokit_client'
+require_relative '../../src/utils/warehouse/github/issues_format'
+
+module Implementation
+  ##
+  # Implementation::FetchIssuesFromGithub
+  #
+  # This class implements a bot that fetches issues from all repositories in a GitHub organization
+  # and saves them into shared storage (e.g., PostgresDB).
+  #
+  # <b>Usage Example</b>
+  #
+  #   read_options = {
+  #     connection: Config::CONNECTION,
+  #     db_table: 'warehouse_sync',
+  #     avoid_process: true,
+  #     where: 'archived=$1 AND tag=$2 ORDER BY inserted_at DESC',
+  #     params: [false, 'FetchIssuesFromGithub']
+  #   }
+  #
+  #   write_options = {
+  #     connection: Config::CONNECTION,
+  #     db_table: 'warehouse_sync',
+  #     tag: 'FetchIssuesFromGithub'
+  #   }
+  #
+  #   options = {
+  #     private_pem: Config::GITHUB_PRIVATE_PEM,
+  #     app_id: Config::GITHUB_APP_ID,
+  #     organization: Config::KOMMITERS_ORGANIZATION
+  #   }
+  #
+  #   shared_storage = Bas::SharedStorage::Postgres.new({ read_options:, write_options: })
+  #
+  #   Implementation::FetchIssuesFromGithub.new(options, shared_storage).execute
+  #
+  class FetchIssuesFromGithub < Bas::Bot::Base
+    PER_PAGE = 100
+
+    # Orchestrates the fetching and formatting of records from GitHub.
+    #
+    def process
+      client_response = initialize_client
+      return client_response if client_response[:error]
+
+      last_run_timestamp = fetch_last_run_timestamp
+
+      client = client_response[:client]
+      repositories = fetch_repositories(client)
+      all_issues = fetch_all_issues(client, repositories, last_run_timestamp)
+
+      { success: { type: 'github_issue', content: all_issues } }
+    end
+
+    # Orchestrates the writing of processed data to the configured shared storage.
+    #
+    def write
+      content = process_response.dig(:success, :content) || []
+      return if content.empty?
+
+      paginate_and_write(content)
+    end
+
+    private
+
+    # Fetches the timestamp of the last successful run from shared storage.
+    def fetch_last_run_timestamp
+      read_result = @shared_storage_reader.read
+      return if read_result.inserted_at.nil?
+
+      Time.parse(read_result.inserted_at.to_s)
+    end
+
+    # Initializes the Octokit client and handles authentication.
+    def initialize_client
+      client_response = Utils::Github::OctokitClient.new(client_params).execute
+      return client_response if client_response[:error]
+
+      client = client_response[:client]
+      client.auto_paginate = true
```

Comment:
In the `fetch_all_issues`, you are setting the auto_paginate to false. Can we remove it from here?